build_docker.sh

```diff
@@ -133,3 +141,6 @@ else
 	echo "Not pushing to dockerhub"
 fi
 
+cd ${ORIGINAL_WORKING_DIRECTORY}
+rm -Rf ${STAGING_DIR}/${STAGING_CLONE_DIR}
```

Comment:
This line scares me... 

if $STAGING_DIR and $STAGING_CLONE_DIR are unset, you're just doing a `sudo rm -rf /`  which is bad...

Could you add a guard that only runs this if `STAGING_DIR` is set?